### PR TITLE
PYTHON-4965 - Consolidate startup and teardown tasks

### PR DESCRIPTION
**Pull request by NoahStapp**

### CODE REVIEW

File: .evergreen/config.yml [hunk 1]
```diff
@@ -1640,6 +1594,7 @@ tasks:
           params:
             binary: bash
             working_dir: src
+            include_expansions_in_env: ["PYTHON_BINARY"]
```
Comment (Jibola) [inline .evergreen/config.yml:1597]: Is this a bug you caught during the fix?

**Review by Jibola — APPROVED** [review]
LGTM.

**Comment by blink1073** [comment]
Docker teardown is handled in [drivers-evergreen-tools](https://github.com/mongodb-labs/drivers-evergreen-tools/blob/master/.evergreen/teardown.sh#L29).  I'd suggest a single teardown script that runs the appropriate drivers-evergreen-tools teardowns: https://github.com/mongodb/mongo-go-driver/blob/master/.evergreen/config.yml#L143

File: .evergreen/config.yml (hunk 1 above)
Comment (NoahStapp) [reply .evergreen/config.yml:1607]: We need to pass this explicitly where needed now to use the correct Python binary. Part of our later work will be to unify the use of `PYTHON_BINARY` to remove the need to pass it explicitly.

**Comment by blink1073** [comment]
[.evergreen/scripts/teardown-docker.sh](https://github.com/mongodb/mongo-python-driver/pull/2017/files#diff-3f85f7d3bea43139f98a5c8a59562ce6438a0e2f076f76e0b0e2e37c708d0ecc) can be removed now.

**Comment by NoahStapp** [comment]
> [.evergreen/scripts/teardown-docker.sh](https://github.com/mongodb/mongo-python-driver/pull/2017/files#diff-3f85f7d3bea43139f98a5c8a59562ce6438a0e2f076f76e0b0e2e37c708d0ecc) can be removed now.

As can https://github.com/mongodb/mongo-python-driver/pull/2017/files#diff-fa520e855fc44f5c073c5ddfca5ac589f408d6174e2b77946e91c3a1323c35ff!

**Review by blink1073 — APPROVED** [review]
LGTM!

**Pull request merged**